extensions/hibernate-validator/runtime/src/main/java/io/quarkus/hibernate/validator/runtime/jaxrs/ResteasyReactiveContextLocaleResolver.java

```diff
@@ -18,6 +18,11 @@ public ResteasyReactiveContextLocaleResolver(HttpHeaders headers) {
 
     @Override
     protected HttpHeaders getHeaders() {
-        return headers;
+        try {
+            headers.getLength(); // this forces the creation of the actual object which will fail if there is no request in flight
+            return headers;
+        } catch (IllegalStateException e) {
+            return null;
```

Comment:
It's not possible to return an empty header object? I'm not sure I understand how the case of not having a request can happen and when the user will be confronted to it?

Reply:
What I understand from the OP's case is that `ResteasyReactiveRequestContext.getHttpHeaders()` fails somehow. I'm wondering if we should return empty headers there and thus cover this case entirely.

Note that I have absolutely no knowledge of the RR codebase so I'm just thinking out loud, given the pattern above looks hackyish and fixing the issue at the source might be a more general solution.

Reply:
`ResteasyReactiveRequestContext.getHttpHeaders()` would indeed make sense to return an empty headers in this case. However it does not make sense for othes in `org.jboss.resteasy.reactive.server.injection.ContextProducers`, so we need a consistent approach